### PR TITLE
Fix typos in codebase

### PR DESCRIPTION
**Pull request by hectormz**
This PR fixes typos in `aesara` with the help of [codespell](https://github.com/codespell-project/codespell).

There was no Issue preceding this PR. There are may be other typos still in the `aesara` that `codespell` didn't pick up, or I was unsure of.

Most typos are in comments and docs, but there are a few in variable names

### CODE REVIEW

File: aesara/sparse/basic.py [hunk 1]
```diff
@@ -3466,7 +3466,7 @@ def make_node(self, x, y):
             raise TypeError(x)
 
         # These are the conversions performed by scipy.sparse.dot
-        if x.type.format == "csc" or x.type.format == "coo":
+        if x.type.format == "csc" or x.type.format == "coup":
```
Comment (ColCarroll) [inline aesara/sparse/basic.py:3469]:
```suggestion
        if x.type.format == "csc" or x.type.format == "coo":
```

Comment (ColCarroll) [reply aesara/sparse/basic.py:3469]: (see https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.coo_matrix.html#scipy.sparse.coo_matrix)

**Review by ColCarroll — APPROVED** [review]
thanks! I checked these reasonably carefully and they seem to really increase readability. I'd suggest not adding too much more to this PR as it can be costly to review (though a followup would be fine if you have more ideas!)

File: aesara/sparse/basic.py (hunk 1 above)
Comment (hectormz) [reply aesara/sparse/basic.py]: Thanks for the fix! I manually blocked this correction elsewhere, but this snuck by me.

**Review by michaelosthege — DISMISSED** [review]
LGTM!

**Comment by hectormz** [comment]
@ColCarroll this branch is now up to date with master and I added a few more fixes

**Comment by brandonwillard** [comment]
I've resolved the merge conflicts on this PR, so, if you want to do any further work on this branch, you'll need to pull those changes.  Also, we've changed the `master` branch to `main`, so be sure to update your local repository accordingly.

**Comment by hectormz** [comment]
Great, thanks @brandonwillard no more typos for now, but can do another sweep in the future. Thanks for the update on `main`

**Comment by twiecki** [comment]
Thanks @hectormz!

**Comment by michaelosthege** [comment]
And thanks @brandonwillard for rebasing it!
Good to see this PR merged :)

**Pull request merged**